src/plugins/data/server/index_patterns/index_patterns_service.ts

```diff
@@ -55,29 +59,37 @@ export class IndexPatternsService implements Plugin<void, IndexPatternsServiceSt
   }
 
   public start(core: CoreStart, { fieldFormats, logger }: IndexPatternsServiceStartDeps) {
-    const { uiSettings } = core;
+    this.#uiSettings = core.uiSettings;
+    this.#fieldFormats = fieldFormats;
+    this.#logger = logger;
 
     return {
-      indexPatternsServiceFactory: async (
-        savedObjectsClient: SavedObjectsClientContract,
-        elasticsearchClient: ElasticsearchClient
-      ) => {
-        const uiSettingsClient = uiSettings.asScopedToClient(savedObjectsClient);
-        const formats = await fieldFormats.fieldFormatServiceFactory(uiSettingsClient);
+      indexPatternsServiceFactory: this.createIndexPatternsService.bind(this),
+    };
+  }
 
-        return new IndexPatternsCommonService({
-          uiSettings: new UiSettingsServerToCommon(uiSettingsClient),
-          savedObjectsClient: new SavedObjectsClientServerToCommon(savedObjectsClient),
-          apiClient: new IndexPatternsApiServer(elasticsearchClient),
-          fieldFormats: formats,
-          onError: (error) => {
-            logger.error(error);
-          },
-          onNotification: ({ title, text }) => {
-            logger.warn(`${title} : ${text}`);
-          },
-        });
+  public async createIndexPatternsService(
+    savedObjectsClient: SavedObjectsClientContract,
+    elasticsearchClient: ElasticsearchClient
+  ) {
+    if (!this.#uiSettings) throw new Error('UI Settings not set in IndexPatternsService.');
```

Comment:
I don't think they can be passed in into this function. Or at least they should not be. My understanding of this `IndexPatternsService` is that it collects the "setup" `core` object and then "start" `core` object and `fieldFormats` with `logger` so that as to serve as a factory later on. So that later `createIndexPatternsService(...` can be called with only two parameters&mdash;`savedObjectsClient` and `elasticsearchClient`.

Reply:
Maybe we should talk about this face to face. I don't understand why these changes are necessary or how they relate to the PR.